src/hotspot/cpu/x86/x86.ad

```diff
@@ -2553,13 +2620,13 @@ instruct ShouldNotReachHere() %{
 
 // =================================EVEX special===============================
 
-instruct setMask(rRegI dst, rRegI src) %{
+instruct setMask(rRegI dst, rRegI src, kReg_K1 mask) %{
```

Comment:
Why can't other instructions destroy `k1` by getting `k1` assigned as a temp? Currenlty, `kReg` includes the whole range of k-registers (k0-k7).

Reply:
I think post-loop vectorization handling currently expects mask to be present in K1, I just pulled it out till the instruction level,  RA allocates opmask registers in reverse order i.e. K7-K1, thus temps should all be allocated K7 registers first.

Reply:
It is just a coincidence. Considering there's a way to produce an arbitrary number of predicates using `VectorMask`, I'd prefer to see a more robust solution that won't cause us problems in the future.
Please, exclude `k1` from `kReg` then.

Reply:
For the time being I have removed K1 from allocable set, till we fix post-loop multi-versioning.

Reply:
Looks good. Please, put comments before `setMask`, `jmpLoopEnd_and_restoreMask`, and `jmpLoopEndU_and_restoreMask` describing why it is safe to use `k1` there.